relay-server/src/services/buffer/stack_provider/mod.rs

```diff
@@ -1,15 +1,45 @@
-use crate::{Envelope, EnvelopeStack};
+use crate::services::buffer::envelope_store::EnvelopeProjectKeys;
+use crate::EnvelopeStack;
+use hashbrown::HashSet;
+use std::future::Future;
 
 pub mod memory;
 pub mod sqlite;
 
+/// State of the initialization of the [`StackProvider`].
+///
+/// This state is necessary for initializing resources whenever a [`StackProvider`] is used.
+#[derive(Debug)]
+pub struct InitializationState {
+    pub envelopes_projects_keys: HashSet<EnvelopeProjectKeys>,
+}
```

Comment:
Is this type really necessary? Shouldn't we just return the HashSet from `StackProvider::initialize`?

Reply:
It was done on purpose to encapsulate an arbitrary state type, I didn't want to have the trait directly expose those keys.